src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -419,9 +481,50 @@ static term start_ap(term ap_config, GlobalContext *global)
         }
     }
 
+    if (term_is_invalid_term(hostname_term)) {
```

Comment:
This code feels like a copy & paste of another block, can we factor out this?

Reply:
Indeed. This has been refactored.